drivers/bluetooth/hci/slz_hci.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+
+#include <sl_btctrl_linklayer.h>
+#include <sl_hci_common_transport.h>
+#include <pa_conversions_efr32.h>
+#include <sl_bt_ll_zephyr.h>
+
+#define SL_BT_CONFIG_ACCEPT_LIST_SIZE	1
+#define SL_BT_CONFIG_MAX_CONNECTIONS	1
+#define SL_BT_CONFIG_USER_ADVERTISERS	1
+#define SL_BT_CONTROLLER_BUFFER_MEMORY  CONFIG_BT_SILABS_HCI_BUFFER_MEMORY
+#define SL_BT_SILABS_LL_STACK_SIZE	1024
+
+static K_KERNEL_STACK_DEFINE(slz_ll_stack, SL_BT_SILABS_LL_STACK_SIZE);
+static struct k_thread slz_ll_thread;
+
+void rail_isr_installer(void)
+{
+	IRQ_CONNECT(RDMAILBOX_IRQn, 0, RDMAILBOX_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_SEQ_IRQn, 0, RAC_SEQ_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_RSM_IRQn, 0, RAC_RSM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(PROTIMER_IRQn, 0, PROTIMER_IRQHandler, NULL, 0);
+	IRQ_CONNECT(MODEM_IRQn, 0, MODEM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(FRC_IRQn, 0, FRC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(BUFC_IRQn, 0, BUFC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(AGC_IRQn, 0, AGC_IRQHandler, NULL, 0);
+}
+
+/**
+ * @brief Transmit HCI message using the currently used transport layer.
+ * The HCI calls this function to transmit a full HCI message.
+ * @param[in] data Data buffer to be transmitted.
+ * @param[in] len Length of the data.
+ * @return 0 - on success, or non-zero on failure.
+ */
+uint32_t hci_common_transport_transmit(uint8_t *data, int16_t len)
+{
+	struct net_buf *buf;
+
+	switch (data[0]) {
+	case h4_event:
+		uint8_t flags, evt = data[1];
+
+		buf = bt_buf_get_evt(evt, false, K_FOREVER);
+		net_buf_add_mem(buf, data + 1, len - 1);
+		flags = bt_hci_evt_get_flags(evt);
+
+		if (flags & BT_HCI_EVT_FLAG_RECV_PRIO) {
+			bt_recv_prio(buf);
+		} else {
+			bt_recv(buf);
+		}
+
+		break;
+	case h4_acl:
+		buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_FOREVER);
+		net_buf_add_mem(buf, data + 1, len - 1);
+		bt_recv(buf);
+		break;
+	default:
+		break;
```

Comment:
It seems to me like unexpected values in data[0] will get swallowed silently, returning zero (success).

How about an error message for such values, and returning a nonzero value?

Reply:
Added an error message printing (through the log subsystem) here.